homeassistant/components/ccl/config_flow.py

```diff
@@ -0,0 +1,54 @@
+"""Config flow for CCL Electronics."""
+
+from __future__ import annotations
+
+import secrets
+from typing import Any
+
+from yarl import URL
+
+from homeassistant.components import webhook
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PATH, CONF_PORT, CONF_WEBHOOK_ID
+from homeassistant.helpers.network import get_url
+
+from .const import DOMAIN
+
+
+class CCLConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        self._set_confirm_only()
+
+        if user_input is not None:
+            webhook_id = secrets.token_hex(4)
+
+            url = URL(get_url(self.hass))
+            assert url.host
+
+            host = url.host
+            port = str(url.port)
+            path = webhook.async_generate_path(webhook_id)
```

Comment:
This won't fail if the passed in webhook id is not unique. To avoid collisions, prefix the id with this integration's domain.